### ISSUE TITLE
False positive for `Style/RedundantParentheses` cop

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#14228]
## Describe

```ruby
class XyzService
  def account_today
    service.estimate_value(
      (yield(dates[yesterday.to_s]) || 0 rescue 0),
      (yield(dates[today.to_s])     || 0 rescue 0),
    ).to_i
  end
end
```

After running `rubocop -A`, the syntax of my code was broken:
```ruby
class XyzService
  def account_today
    service.estimate_value(
      yield(dates[yesterday.to_s]) || 0 rescue 0,
      yield(dates[today.to_s])     || 0 rescue 0,
    ).to_i
  end
end
```

## Rubocop

```console
1.75.8 (using Parser 3.3.8.0, rubocop-ast 1.44.1, analyzing as Ruby 2.7, running on ruby 2.7.8) [x86_64-linux]
  - rubocop-capybara 2.22.1
  - rubocop-factory_bot 2.27.1
  - rubocop-migration 0.7.1
  - rubocop-performance 1.25.0
  - rubocop-rails 2.32.0
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.6.0
  - rubocop-rspec_rails 2.31.0
  - rubocop-thread_safety 0.7.2
```
